# Descriptive Statistics Assignment

In [ ]:
import pandas as pd

data_url = "http://data.insideairbnb.com/united-states/ny/new-york-city/2020-06-08/visualisations/listings.csv"

### Read the aggregated AirBnB listings of New York (using `data_url`) into a Pandas data frame.

In [ ]:
df = pd.read_csv(data_url)
df

id  ... availability_365
0          2060  ...              365
1          2595  ...              335
2          3831  ...              276
3          5099  ...                0
4          5121  ...              365
...         ...  ...              ...
49525  43702714  ...              365
49526  43702765  ...              135
49527  43703128  ...                8
49528  43703156  ...               12
49529  43703359  ...              230

[49530 rows x 16 columns]

### Print the data type for each column.

In [ ]:
df.dtypes

id                                  int64
name                               object
host_id                             int64
host_name                          object
neighbourhood_group                object
neighbourhood                      object
latitude                          float64
longitude                         float64
room_type                          object
price                               int64
minimum_nights                      int64
number_of_reviews                   int64
last_review                        object
reviews_per_month                 float64
calculated_host_listings_count      int64
availability_365                    int64
dtype: object

### Generate a basic set of descriptive statistics for the numeric fields in the data set.

In [ ]:
stats = df.describe()
dfNum = df.select_dtypes("number")
stats

id  ...  availability_365
count  4.953000e+04  ...      49530.000000
mean   2.295964e+07  ...        126.666848
std    1.352683e+07  ...        142.381428
min    2.060000e+03  ...          0.000000
25%    1.085050e+07  ...          0.000000
50%    2.233602e+07  ...         79.000000
75%    3.557779e+07  ...        267.000000
max    4.370336e+07  ...        365.000000

[8 rows x 10 columns]

### Calculate the following for each numeric field in the data set and add them to the data frame containing the basic set of descriptive statistics.

- Variance
- Mean Absolute Deviation
- Range
- Interquartile Range

In [ ]:
variance = pd.Series(df.var(), name="variance")
mad = pd.Series(df.mad(), name="mean absolute deviation")
range = pd.Series(dfNum.max()-dfNum.min(), name="range")
iqr = pd.Series(df.quantile(q=0.75)-df.quantile(q=0.25), name="iqr")
stats = stats.append(variance)
stats = stats.append(mad)
stats = stats.append(range)
stats = stats.append(iqr)
stats



id  ...  availability_365
count                    4.953000e+04  ...      49530.000000
mean                     2.295964e+07  ...        126.666848
std                      1.352683e+07  ...        142.381428
min                      2.060000e+03  ...          0.000000
25%                      1.085050e+07  ...          0.000000
50%                      2.233602e+07  ...         79.000000
75%                      3.557779e+07  ...        267.000000
max                      4.370336e+07  ...        365.000000
variance                 1.829752e+14  ...      20272.470950
mean absolute deviation  1.181604e+07  ...        126.553298
range                    4.370130e+07  ...        365.000000
iqr                      2.472729e+07  ...        267.000000

[12 rows x 10 columns]

### Generate a basic set of descriptive statistics for the categorical fields in the data set.

In [ ]:
df.describe(include="object").T

count unique                  top   freq
name                 49512  48247  Home away from home     21
host_name            49524  11572              Michael    416
neighbourhood_group  49530      5            Manhattan  21963
neighbourhood        49530    222         Williamsburg   3752
room_type            49530      4      Entire home/apt  25716
last_review          38211   2077           2020-01-01    940

### Calculate the Pearson correlation coefficients between price, minimum_nights, reviews_per_month, calculated_host_listings_count and availability_365

In [ ]:
df[['price', 'minimum_nights', 'reviews_per_month', 'calculated_host_listings_count', 'availability_365']].corr()

price  ...  availability_365
price                           1.000000  ...          0.055788
minimum_nights                  0.015923  ...          0.148435
reviews_per_month              -0.011623  ...          0.213093
calculated_host_listings_count  0.004730  ...          0.173181
availability_365                0.055788  ...          1.000000

[5 rows x 5 columns]

### Group the data by neighbourhood and calculate the mean and standard deviation of revenue for each neighbourhood.

In [ ]:
aggs = {'price': ['mean', 'std']}
neighborhoods = df.groupby(['neighbourhood']).agg(aggs)
neighborhoods

price             
                        mean          std
neighbourhood                            
Allerton           94.500000    83.219379
Arden Heights     102.600000    54.834296
Arrochar          104.315789   136.241229
Arverne           187.125000   231.710658
Astoria           104.109865   124.606979
...                      ...          ...
Windsor Terrace   138.067114    89.830596
Woodhaven          69.638095    43.294191
Woodlawn           61.700000    24.882167
Woodrow          2350.000000  2333.452378
Woodside           69.383721    85.328235

[222 rows x 2 columns]

### Create a new mimimum revenue index variable that combines the price, minimum nights, calculate host listings count into a single value for each dwelling. Then sort the data frame by that value to see which dwelling has the most financial strength. 



In [ ]:
df['minimum_revenue_index'] = (df['price'] * df['minimum_nights'] * df['calculated_host_listings_count'])
df.sort_values('minimum_revenue_index', ascending=False).head(10).reset_index().T

0  ...                                                  9
index                                                                      14754  ...                                              33441
id                                                                      13443640  ...                                           31726939
name                            Deluxe 3-BR + 3.5Bath with sweeping city views!!  ...  Terrific Tribeca 1BR, Gym, Roof deck, Indoor p...
host_id                                                                 30283594  ...                                          107434423
host_name                                                   Global Luxury Suites  ...                                         Blueground
neighbourhood_group                                                    Manhattan  ...                                          Manhattan
neighbourhood                                                            Midtown  ...                                            Tribeca
latitude                                                                 40.7514  ...                                            40.7158
longitude                                                               -73.9697  ...                                           -74.0059
room_type                                                        Entire home/apt  ...                                    Entire home/apt
price                                                                       1100  ...                                                389
minimum_nights                                                               120  ...                                                 30
number_of_reviews                                                              0  ...                                                  0
last_review                                                                  NaN  ...                                                NaN
reviews_per_month                                                            NaN  ...                                                NaN
calculated_host_listings_count                                                83  ...                                                280
availability_365                                                             184  ...                                                  0
minimum_revenue_index                                                   10956000  ...                                            3267600

[18 rows x 10 columns]

### Group the data by neighbourhood_group and calculate the average minimum revenue index for each neighbourhood_group.

In [ ]:
aggs = {'minimum_revenue_index': ['mean']}
neighborhoodGroups = df.groupby(['neighbourhood_group']).agg(aggs)
neighborhoodGroups

minimum_revenue_index
                                     mean
neighbourhood_group                      
Bronx                          987.498331
Brooklyn                      3434.846270
Manhattan                    48328.517097
Queens                        7537.764338
Staten Island                 1201.845946

**Conditional Probability Practice**



**Example 1**

It is raining. What is the probability that I will need to go back and get my umbrella before I leave the house?
 
I forget to check the app about 30% of the time. 
There's 95% chance that the app will tell me that it is raining when it's raining outside. 
There's a 30% chance that I will need to go back and get an umbrella because I forgot to check the app. 

There's a 5% chance that I'll have to go back and get the umbrella because it's raining and I checked the app.
There’s a 30% chance that I’ll have to go back and get the umbrella because I forgot to check the app. 
There’s a 35% chance that I’ll have to go back and get the umbrella. 

**Example 2**

The chance of an accident happening at a Mal-Wart store on a regular day is 1%. On Black Friday, the chance is 20%. An accident report was handed in without a date on it, what is the probability the accident happened on Black Friday?

P(Black Friday) = 1/365 - Chance that the day is Black Friday

P(Regular Day) = 364/365 - Chance that the day is a regular day

P(Accident|Regular Day) = 1/100 - Chance of accidents on a regular day

P(Accident|Black Friday) = 1/2 - Chance of accidents on Black Friday

P(Accident) = P(Black Friday) * P(Accident|Black Friday) + P(Regular Day) * P(Accident|Regular Day) - Chance of an accident on any given day

P(Accident) = 1/7300 + 364/36500 - Chance of an accident on any given day

P(Accident) = 369/36500 - Chance of an accident on any given day

P(Black Friday|Accident) = P(Accident|Black Friday) * P(Black Friday) / P(Accident) - Chance that any given accident happened on Black Friday

P(Accident|Black Friday) * P(Black Friday) = 1/5 * 1/365 = 1/1825 - Likelihood * Prior

P(Black Friday|Accident) = (1/1825)/(369/36500) = 20/369 = 5.542% - Chance that any given accident happened on Black Friday  


If there is an accident in the store, there is a 5.542% chance it is Black Friday, which means there is a 94.5% chance the accident did not happen on Black Friday. 

**Example 3**

Toyota produces many parts to build their cars, and all their parts needs to pass an automatic inspection. Based on previous observation, the battery has a 90% chance of being good, 2% chance of being slightly damaged, and 8% chance of fully damaged. The machine that inspects can only recognize the fully damaged batteries. What is the quality of the batteries that make it through the inspection machine and get shipped?
 
The probability of good battery is 90%, so P(G) = .9. the probability of slightly damaged is 2% so P(SD) = .02. lastly the probability of fully damaged battery is 8% so P(FD) = .08. Since we know 92 batteries will get through the machine and two of those are SD and other 90 are good we want to get proportion of good batteries divided by all the batteries that passes. Thus 90/92 = .978, so there is 97.8% chance that the battery that will be shipped are good.
